### PR TITLE
Trains always fully dock within walls. Adds rail signal to station block

### PR DESCRIPTION
**Pull request by Cyberboss**
Untested with oil outposts atm. Give me a few hours

Too large: https://media.discordapp.net/attachments/198104144391700490/449647174532464640/blueprint.png

### CODE REVIEW

**Comment by demipixel** [comment]
I'm curious what you meant by "too large"?

**Comment by Cyberboss** [comment]
Github wouldn't load the image with markdown :P

**Comment by demipixel** [comment]
Ahhh, alright. Is this ready for merging?

**Comment by Cyberboss** [comment]
Not quite, oil outposts are [kinda screwy](https://cdn.discordapp.com/attachments/198104144391700490/449689320933294104/blueprint.png)

**Comment by Cyberboss** [comment]
Here we go

File: tools/oilOutpost.js [hunk 1]
```diff
@@ -311,18 +311,21 @@ module.exports = function(string, opt = {}) {
   let trainStopLocation = null;
 
   if (INCLUDE_TRAIN_STATION) {
-    trainStopLocation = generateTrainStation(bp, { x: target.x + 3 + 3 * TANKS, y: target.y - 2 }, Math.max(bp.bottomRight().y, target.y - 2 -
-      WAGONS * 7), {
+    trainStopLocation = generateTrainStation(bp,
+       { x: target.x + 3 + 3 * TANKS, y: target.y - 2 },
+        Math.max(bp.bottomRight().y, target.y + 2 + WAGONS * 7 + ((SINGLE_HEADED_TRAIN ? 0 : 1) * LOCOMOTIVES)),
```
Comment (Cyberboss) [inline tools/oilOutpost.js:316]: This seemed to be the issue with oil lines being too short

Comment (demipixel) [reply tools/oilOutpost.js:316]: Trains have been the biggest headache of all the issues in this project :P

**Comment by demipixel** [comment]
Awesome :) Although if it's double headed, there should be two rail signals, right?

**Comment by Cyberboss** [comment]
Correct, but I decided to leave that to the builder because one never knows if they want chain or standard signals

**Comment by demipixel** [comment]
You seem better at this than I am :P Should I merge?

**Comment by Cyberboss** [comment]
Yep!

**Comment by demipixel** [comment]
Alright, I'll add it to Autotorio soon!

**Pull request merged**